packages/flutter/test/widgets/page_view_test.dart

```diff
@@ -1372,4 +1372,35 @@ void main() {
       await tester.pumpWidget(createPageView(null));
     });
   });
+
+  testWidgets('Get the page value before the content dimension is determined,do not throw an assertion and return null', (WidgetTester tester) async {
+    // Regression test for https://github.com/flutter/flutter/issues/146986.
+    final PageController controller = PageController();
+    late String currentPage;
+    addTearDown(controller.dispose);
+    await tester.pumpWidget(MaterialApp(
+      home: Material(
+        child: PageView(
+          controller: controller,
+          children: <Widget>[
+            Builder(
+              builder: (BuildContext context) {
+                currentPage = controller.page == null ? 'null' : 'not empty';
+                return Center(child: Text(currentPage));
+              },
+            ),
+          ],
+        ),
+      ),
+    ));
+    expect(find.text('null'), findsOneWidget);
+    expect(find.text('not empty'), findsNothing);
+    expect(currentPage, 'null');
+
+    await tester.pump();
+    currentPage = controller.page == null ? 'null' : 'not empty';
+    expect(find.text('not empty'), findsOneWidget);
```

Comment:
Hmm, actually it looks like we may have found a bug in adjusting this test. @hello-coder-xu can you take a look?

Reply:
Reason for test failure: The page is not refreshed after `await tester.pump()`, and text is still displayed as `null`. The view needs to be updated before the test can pass normally.